source/vecs/entity.d

```diff
@@ -1313,46 +1313,41 @@ unittest
 {
 	auto em = new EntityManager();
 
-	foreach (i; 0..10) em.gen!(Foo, Bar);
+	foreach (i; 0..10) em.entity().add!(Foo, Bar);
 
 	assertEquals(10, em.size!Foo());
-	em.removeAll!Foo();
+	em.clear!Foo();
 	assertEquals(0, em.size!Foo());
 }
 
 @system
-@("entity: EntityManager: set")
+@("entity: EntityManager: setComponent")
 unittest
 {
 	auto em = new EntityManager();
 
-	auto e = em.gen();
-	assertTrue(em.set(e, Foo(4, 5)));
-	assertThrown!AssertError(em.set(Entity(0, 5), Foo(4, 5)));
-	assertThrown!AssertError(em.set(Entity(2), Foo(4, 5)));
+	auto e = em.entity();
+	assertTrue(em.setComponent(e, Foo(4, 5)));
+	assertThrown!AssertError(em.setComponent(Entity(0, 5), Foo(4, 5)));
+	assertThrown!AssertError(em.setComponent(Entity(2), Foo(4, 5)));
 	assertEquals(Foo(4, 5), *em.storageInfoMap[ComponentId!Foo].get!(Foo).get(e));
 
 	{
-		auto components = em.set(em.gen(), Foo(4, 5), Bar("str"));
+		auto components = em.setComponent(em.entity(), Foo(4, 5), Bar("str"));
 		assertEquals(Foo(4,5), *components[0]);
 		assertEquals(Bar("str"), *components[1]);
 	}
 
 	{
-		auto components = em.set!(Foo, Bar, int)(em.gen());
+		auto components = em.addComponent!(Foo, Bar, int)(em.entity());
 		assertEquals(Foo.init, *components[0]);
 		assertEquals(Bar.init, *components[1]);
 		assertEquals(int.init, *components[2]);
 	}
 
-	assertThrown!AssertError(em.set!Foo(Entity(45)));
-	assertThrown!AssertError(em.set!(Foo, Bar)(Entity(45)));
-	assertThrown!AssertError(em.set(Entity(45), Foo.init, Bar.init));
-
-	assertFalse(__traits(compiles, em.set!(Foo, Bar, int, Bar)(em.gen())));
-	assertFalse(__traits(compiles, em.set(em.gen(), Foo(4, 5), Bar("str"), Foo.init)));
-	assertFalse(__traits(compiles, em.set!(Foo, Bar, InvalidComponent)(em.gen())));
-	assertFalse(__traits(compiles, em.set!(InvalidComponent)(em.gen())));
+	assertThrown!AssertError(em.addComponent!Foo(Entity(45)));
```

Comment:
This is supposed to pass in `release`. I won't be bothered with `betterC` yet, it would be a cool feature, but it is not my focus right now. The plan is to first improve the lib overall, then support `nogc` and then support `multithread`.

Reply:
What's the advantage of unit testing in release mode? And there is a better option for this specific case:
```d
version(assert)
assertThrown!AssertError(...);
```

This version modifier evaluates to true if checks are being emitted for AssertExpressions.

Reply:
fixed